test/stdcpp/Makefile

```diff
@@ -0,0 +1,19 @@
+include ../common.mak
+
+TESTS:=array
+
+.PHONY: all clean
+all: $(addprefix $(ROOT)/,$(addsuffix .done,$(TESTS)))
+
+$(ROOT)/%.done : $(ROOT)/%
+	@echo Testing $*
+	$(QUIET)$(TIMELIMIT)$(ROOT)/$* $(RUN_ARGS)
+	@touch $@
+
+$(ROOT)/%: $(SRC)/%_test.d $(SRC)/%.cpp
+	mkdir -p $(dir $@)
+	$(QUIET)$(CXX) $(CXXFLAGS) -std=c++11 -c -o $(ROOT)/$*_cpp.o $(SRC)/$*.cpp
```

Comment:
Sorry as in 
```make
OS := $(shell uname)
CXXFLAGS :=  -std=c++11
ifeq ($(OS),Darwin)
    CXX_FLAGS += -stdlib=libc++
endif
...
$(ROOT)/%: $(SRC)/%_test.d $(SRC)/%.cpp
 	mkdir -p $(dir $@)
        $(QUIET)$(CXX) $(CXXFLAGS) -c -o $(ROOT)/$*_cpp.o $(SRC)/$*.cpp
```

Reply:
I think `CXXFLAGS` should probably be initially defined in the root makefile. It seems weird to put it here in a leaf...

Reply:
it also needs to aggregate `CFLAGS`

Reply:
Other makefiles use `ifeq ($(OS),osx)` instead of `Darwin`... I can't test this. It's fiddly to get right.